app/src/androidTest/java/org/mozilla/focus/activity/URLCompletionTest.java

```diff
@@ -67,4 +91,206 @@ public void CompletionTest() throws InterruptedException, UiObjectNotFoundExcept
     }
+
+    @Test
+    public void CustomCompletionTest() throws InterruptedException, UiObjectNotFoundException {
+        // add custom urls in settings
+        // Enable custom autocomplete, and add 2 URLs
+        // Open Settings->Autocomplete menu
+        OpenCustomCompleteDialog();
+
+        // Enable URL autocomplete, and tap Custom URL add button
+        toggleAutocomplete();
+        addAutoComplete("680news.com");
+        Espresso.pressBack();
+        Espresso.pressBack();
+        Espresso.pressBack();
+
+        // Check for custom autocompletion
+        TestHelper.inlineAutocompleteEditText.waitForExists(waitingTime);
+        TestHelper.inlineAutocompleteEditText.setText("68");
+        TestHelper.hint.waitForExists(waitingTime);
+        assertTrue (TestHelper.inlineAutocompleteEditText.getText().equals("680news.com"));
+        TestHelper.cleartextField.click();
+
+        // Remove custom autocompletion site
+        OpenCustomCompleteDialog();
+        removeACSite();
+        Espresso.pressBack();
+        Espresso.pressBack();
+        Espresso.pressBack();
+
+        // Check autocompletion
+        TestHelper.inlineAutocompleteEditText.waitForExists(waitingTime);
+        TestHelper.inlineAutocompleteEditText.setText("68");
+        TestHelper.hint.waitForExists(waitingTime);
+        assertTrue (TestHelper.inlineAutocompleteEditText.getText().equals("68"));
+        TestHelper.cleartextField.click();
+
+        // add custom autocompletion site, but disable autocomplete
+        OpenCustomCompleteDialog();
+        addAutoComplete("geckoview.com");
+        Espresso.pressBack();
+        toggleAutocomplete();       // Disable autocomplete
+        Espresso.pressBack();
+        Espresso.pressBack();
+
+        // Check autocompletion
+        TestHelper.inlineAutocompleteEditText.waitForExists(waitingTime);
+        TestHelper.inlineAutocompleteEditText.setText("gecko");
+        TestHelper.hint.waitForExists(waitingTime);
+        assertTrue (TestHelper.inlineAutocompleteEditText.getText().equals("gecko"));
+        TestHelper.cleartextField.click();
+
+        // Now enable autocomplete
+        OpenCustomCompleteDialog();
+        toggleAutocomplete();
+        Espresso.pressBack();
+        Espresso.pressBack();
+
+        // Check autocompletion
+        TestHelper.inlineAutocompleteEditText.waitForExists(waitingTime);
+        TestHelper.inlineAutocompleteEditText.setText("gecko");
+        TestHelper.hint.waitForExists(waitingTime);
+        assertTrue (TestHelper.inlineAutocompleteEditText.getText().equals("geckoview.com"));
+        TestHelper.cleartextField.click();
+
+        // Cleanup
+        OpenCustomCompleteDialog();
+        removeACSite();
+        Espresso.pressBack();
+        toggleAutocomplete();       // Disable autocomplete
+        Espresso.pressBack();
+        Espresso.pressBack();
+    }
+
+    @Test
+    public void DuplicateACSiteTest() throws UiObjectNotFoundException {
```

Comment:
\[PMD\] INFO: The method 'DuplicateACSiteTest\(\)' has a NCSS line count of 24\.